js/store.js

```diff
@@ -112,51 +216,139 @@ const store = {
     }
 
     this._write(websites);
+  },
+
+  isNewWebsite(hostname) {
+
+    if (!this._websites || !this._websites[hostname]) {
+      return true;
+    }
+    return false;
+  },
 
-    if (newSite) {
-      this.updateChild(this.outputWebsite(hostname, websites[hostname]));
+  getHostnameVariants(hostname) {
+    const hostnameVariants = [hostname];
+    const hostnameArr = hostname.split('.');
+    const numDots = hostnameArr.length - 1;
+    for (let i = 0; i < numDots - 1; i++) {
+      hostnameArr.shift();
+      hostname = hostnameArr.join('.');
+      hostnameVariants.push(hostname);
     }
+    return hostnameVariants;
+  },
+
+  // check if third party is on the whitelist (owned by the first party)
+  // returns true if it is and false otherwise
+  onWhiteList(firstPartyFromRequest, thirdPartyFromRequest) {
+    if (thirdPartyFromRequest && this.firstPartyWhiteList) {
+      const hostnameVariantsFirstParty
+        = this.getHostnameVariants(firstPartyFromRequest);
+      for (let i = 0; i < hostnameVariantsFirstParty.length; i++) {
+        if (this.firstPartyWhiteList
+          .hasOwnProperty(hostnameVariantsFirstParty[i])) {
+          // first party is in the whitelist
+          const index = this.firstPartyWhiteList[hostnameVariantsFirstParty[i]];
+          const hostnameVariantsThirdParty
+            = this.getHostnameVariants(thirdPartyFromRequest);
+          for (let j = 0; j < hostnameVariantsThirdParty.length; j++) {
+            if (this.thirdPartyWhiteList[index]
+              .includes(hostnameVariantsThirdParty[j])) {
+              return true;
+            }
+          }
+          return false;
+        }
+      }
+    }
+    return false;
   },
 
   setFirstParty(hostname, data) {
     if (!hostname) {
       throw new Error('setFirstParty requires a valid hostname argument');
     }
+    const isNewWebsite = this.isNewWebsite(hostname);
+
     this.setWebsite(hostname, data);
+
+    if (isNewWebsite) {
+      this.updateChild(this.outputWebsite(hostname, this._websites[hostname]));
+    }
   },
 
+  // @todo use data from capture to update UI/filter graph
+  // eslint-disable-next-line no-unused-vars
```

Comment:
The way `setThirdParty` is set up right now, I don't actually use the request data, just the `origin` and `target` hostname strings. As a reminder, the request data received from capture is:
```javascript
const data = {
        document: documentUrl.hostname,
        target: targetUrl.hostname,
        origin: originUrl.hostname,
        requestTime: response.timeStamp
      };
```
This data can be merged into the `thirdParty` object, but these values would be overwritten as soon as another unwhitelisted site makes a request to this third party. I figured this would be addressed when we refactor storage and complete #65 . What do you think should be done in the interim?